### DISCUSSION TITLE
Mainboard alternative

### DISCUSSION THREAD

**Discussion by A-Smour** [jyjblrd/Core-R-Theta-4-Axis-Printer#7] [Q&A]
Can I use the MKS Gen-L v2.1 instead of the Fly-E3-Pro-V3?

**Comment by jyjblrd** [comment]
yep, as long as it is supported by reprap firmware

**Reply by askpluscreations** [reply]
CAN YOU DETAIL ON how to compile and load the firmware also the wiring diagrma will be helpful

**Reply by oldnslow** [reply]
@askpluscreations , see [this post](https://github.com/jyjblrd/Core-R-Theta-4-Axis-Printer/discussions/47?sort=new#discussioncomment-13298513) in the [reprap firmware config discussion](https://github.com/jyjblrd/Core-R-Theta-4-Axis-Printer/discussions/47) for link to a wiki with information on how to obtain and setup reprapfirmware for the Core-R-&theta; and also useful links for setting up and using reprapfirmware.

**Comment by iyxx37** [comment]
FLY E3 Pro Ultra RRF( TMC2209)， is this ok for mainboard

**Reply by oldnslow** [reply]
Yes, although the [Team Gloomy RRF supported board list](https://teamgloomy.github.io/supported_boards.html) has not been updated yet, the Fly E3 Pro Ultra does show up in their [Fly/Mellow Board section](https://teamgloomy.github.io/fly_e3_ultra_general_3_5.html).

Note also getting the **ESP32 Board for RRF** option is required to have ethernet/wifi connectivity in order to run headless and use Duet Web Control as recommended.

**Comment by iyxx37** [comment]
![1111PPPP](https://github.com/user-attachments/assets/13a3af21-f7cd-4267-aeac-3746130981ee)
and which point for the head fan ?   which for the heater of head and bed heater,  and which for these four motor one by one ?

**Reply by the-Geeson** [reply]
https://teamgloomy.github.io/fly_e3_pro_v3_pins.html

**Reply by oldnslow** [reply]
Also [see this post](https://github.com/jyjblrd/Core-R-Theta-4-Axis-Printer/discussions/47?sort=new#discussioncomment-13298513) for link to wiki with more detail on how drive and fan mapping is determined.

**Comment by iyxx37** [comment]
if i replace to [Duet 3  6HC mianboard , could it will improve the speed and precise than the fly e3 pro ultra ?

**Reply by oldnslow** [reply]
My speculation is that there would be very little, if any, discernible difference in the motion control performance between the Duet and Mellow boards as applied to the Core-R-&theta;.  

I think at this stage in its development, the biggest opportunities for improving the speed and precision of the Core-R-&theta; are in improving the mechanical design, particularly the rigidity of the B-axis.   Also, of course, ensuring good alignment of all the axes during the build process will be important.

So if financial costs are a consideration, i would suggest that putting the price  difference between the Duet and Mellow boards into mechanical components would be a better investment.

**Reply by iyxx37** [reply]
thank you verymuch

**Reply by iyxx37** [reply]
B axle is the rotare of head axlie ?

**Reply by oldnslow** [reply]
The B-axis rotates the extruder.  See [video linked in repository README at 6:13](https://youtu.be/VEgwnhLHy3g?t=373).
